src/bun.js/webcore/streams.zig

```diff
@@ -524,7 +524,7 @@ pub const Result = union(Tag) {
                 promise.resolve(globalThis, JSValue.jsBoolean(false));
             },
             else => {
-                const value = result.toJS(globalThis);
+                const value = result.toJS(globalThis) catch .zero; // TODO: properly propagate exception upwards
```

Comment:
I guess this is fine for now? It will still trigger the exception checker if result is temporary or temporary_and_done so it can be caught in the future

Reply:
```suggestion
                const value = result.toJS(globalThis) catch |err| {
                  result.* = .{ .temporary = .{} };
                  promise.reject(globalThis, err);
                  return;
                };
```